### ISSUE TITLE
Not able to start the racnode1 : Error in grid setup 

### ISSUE THREAD

**Issue by onkarnigam14** [oracle/docker-images#1596]
Hi ,

I am facing issue in grid setup : 

04-28-2020 07:02:16 UTC :  : GRID_RESPONSE_FILE env variable set to empty. configGrid.sh will use standard cluster responsefile
04-28-2020 07:02:16 UTC :  : Location for User script SCRIPT_ROOT set to /common_scripts
04-28-2020 07:02:16 UTC :  : IGNORE_CVU_CHECKS is set to true
04-28-2020 07:02:16 UTC :  : Oracle SID is set to ORCLCDB
04-28-2020 07:02:16 UTC :  : Oracle PDB name is set to ORCLPDB
04-28-2020 07:02:16 UTC : : Check failed for network card for  for public IP 172.16.1.150
04-28-2020 07:02:16 UTC : : Error has occurred in Grid Setup, Please verify!

**Comment by onkarnigam14** [comment]
When i login to the container, I found this issue : 

GRID_SETUP_FILE=setupGrid.sh
CMAN_IP=172.16.1.15
_=/usr/sbin/oracleinit
Trying to run as user instance, but $XDG_RUNTIME_DIR is not set.

**Comment by psaini2018** [comment]
@onkarnigam14 

Are you trying to setup cluster using response file? Can you point me to the README.md which you used to setup the cluster?

Also, please provide following:
```
docker info
systemctl status docker
systemctl status firewalld
systemctl status  
uname -r
getenforce
```

Login to container and provide me following:
```
ps -u grid
ps -u oracle
```

**Comment by onkarnigam14** [comment]
Hi I have used the RAC setup git hub repository to do it : https://github.com/oracle/docker-images/tree/master/OracleDatabase/RAC
Have installed all the three images that have been mentioned in the image :

**Comment by onkarnigam14** [comment]
docker info
systemctl status docker
systemctl status firewalld
systemctl status  
uname -r
getenforce

===============================================================================================================================
[root@ip-172-31-47-145 ~]# docker info
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 36
 Server Version: 19.03.1-ol
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: c4446665cb9c30056f4998ed953e6d4ff22c7c39
 runc version: 4bb1fe4ace1a32d3676bb98f5d3b6a4e32bf6c58
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.14.35-1902.301.1.el7uek.x86_64
 Operating System: Oracle Linux Server 7.8
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 31.15GiB
 Name: ip-172-31-47-145.ap-south-1.compute.internal
 ID: 72PO:CQI5:NXQW:7VK3:OXBF:L57N:JLAM:PDQ5:TMA3:K4Z4:HTN7:E5LT
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

===============================================================================================================================

[root@ip-172-31-47-145 ~]# systemctl status docker
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2020-04-29 03:23:15 UTC; 1h 1min ago
     Docs: https://docs.docker.com
 Main PID: 937 (dockerd)
    Tasks: 17
   Memory: 139.7M
   CGroup: /system.slice/docker.service
           └─937 /usr/bin/dockerd --cpu-rt-runtime=950000 --cpu-rt-period=1000000 --exec-opt=native.cgroupdriver=systemd

Apr 29 03:23:11 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:11.514860688Z" level=info msg="Loading containers: start."
Apr 29 03:23:11 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:11.775714108Z" level=info msg="Default bridge (docker0) is a...ddress"
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.049358048Z" level=error msg="ed3691c7391273a54872a9c30d4c...tainer"
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.049397858Z" level=error msg="Failed to start container ed3691c73...
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.049549713Z" level=info msg="Loading containers: done."
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.122586990Z" level=warning msg="Not using native diff for ...verlay2
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.122797345Z" level=info msg="Docker daemon" commit=ead9442...03.1-ol
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.123310696Z" level=info msg="Daemon has completed initialization"
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal dockerd[937]: time="2020-04-29T03:23:15.139287817Z" level=info msg="API listen on /var/run/docker.sock"
Apr 29 03:23:15 ip-172-31-47-145.ap-south-1.compute.internal systemd[1]: Started Docker Application Container Engine.
Hint: Some lines were ellipsized, use -l to show in full.
[root@ip-172-31-47-145 ~]#

===============================================================================================================================

[root@ip-172-31-47-145 ~]# systemctl status firewalld
Unit firewalld.service could not be found.
[root@ip-172-31-47-145 ~]#


===============================================================================================================================

[root@ip-172-31-47-145 ~]# systemctl status
● ip-172-31-47-145.ap-south-1.compute.internal
    State: running
     Jobs: 0 queued
   Failed: 0 units
    Since: Wed 2020-04-29 03:23:05 UTC; 1h 3min ago
   CGroup: /
           ├─1 /usr/lib/systemd/systemd --switched-root --system --deserialize 21
           ├─user.slice
           │ ├─user-0.slice
           │ │ └─session-2.scope
           │ │   └─1331 /usr/sbin/anacron -s
           │ └─user-1000.slice
           │   └─session-1.scope
           │     ├─1195 sshd: ec2-user [priv
           │     ├─1198 sshd: ec2-user@pts/0
           │     ├─1199 -bash
           │     ├─1218 sudo su -
           │     ├─1220 su -
           │     ├─1221 -bash
           │     ├─1667 systemctl status
           │     └─1668 less
           └─system.slice
             ├─irqbalance.service
             │ └─535 /usr/sbin/irqbalance --foreground
             ├─containerd.service
             │ ├─ 868 /usr/bin/containerd
             │ └─1364 containerd-shim -namespace moby -workdir /var/lib/containerd/io.containerd.runtime.v1.linux/moby/ad92c439c5cae7c2fc42cf4299939a49ffa502f4fd23c60f0
             ├─docker-ad92c439c5cae7c2fc42cf4299939a49ffa502f4fd23c60f03dd9aec2e58b1d3.scope
             │ └─1380 [systemd]
             ├─systemd-udevd.service
             │ └─402 /usr/lib/systemd/systemd-udevd
             ├─docker.service
             │ └─937 /usr/bin/dockerd --cpu-rt-runtime=950000 --cpu-rt-period=1000000 --exec-opt=native.cgroupdriver=systemd
             ├─chronyd.service
             │ └─526 /usr/sbin/chronyd
             ├─systemd-journald.service
             │ └─353 /usr/lib/systemd/systemd-journald
             ├─sshd.service
             │ └─940 /usr/sbin/sshd -D
             ├─crond.service
             │ └─944 /usr/sbin/crond -n
             ├─gssproxy.service
             │ └─556 /usr/sbin/gssproxy -D
             ├─rsyslog.service
             │ └─936 /usr/sbin/rsyslogd -n
             ├─serial_console.service
             │ └─945 /sbin/agetty --keep-baud 115200,38400,9600 serial_console dumb
             ├─rpcbind.service
             │ └─484 /sbin/rpcbind -w
             ├─network.service
             │ └─803 /sbin/dhclient -1 -q -lf /var/lib/dhclient/dhclient--eth0.lease -pf /var/run/dhclient-eth0.pid -H ip-172-31-47-145 eth0
             ├─acpid.service
             │ └─551 /usr/sbin/acpid
             ├─dbus.service
             │ └─523 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
             ├─system-getty.slice
             │ └─getty@tty1.service
             │   └─946 /sbin/agetty --noclear tty1 linux
             └─systemd-logind.service
               └─533 /usr/lib/systemd/systemd-logind
[root@ip-172-31-47-145 ~]#




===============================================================================================================================

               └─533 /usr/lib/systemd/systemd-logind
[root@ip-172-31-47-145 ~]# uname -r
4.14.35-1902.301.1.el7uek.x86_64
[root@ip-172-31-47-145 ~]#



===============================================================================================================================

[root@ip-172-31-47-145 ~]# getenforce
Permissive
[root@ip-172-31-47-145 ~]#


===============================================================================================================================

**Comment by onkarnigam14** [comment]
[grid@racnode1 ~]$ ps -u grid
  PID TTY          TIME CMD
  467 pts/1    00:00:00 bash
  487 pts/1    00:00:00 ps
[grid@racnode1 ~]$ ps -u oracle
  PID TTY          TIME CMD
[grid@racnode1 ~]$ id oracle
uid=54321(oracle) gid=54321(oinstall) groups=54321(oinstall),54322(dba),54323(oper),54324(backupdba),54325(dgdba),54326(kmdba),54330(racdba),54334(asmadmin),54335(asmdba),54336(asmoper)
[grid@racnode1 ~]$

**Comment by psaini79** [comment]
OK. Your setup seems to be correct except you are using systemd cgroup driver but that should not be an issue at this moment.

Please execute following and provide following output:
```
docker exec -i -t racnode1 /bin/bash
ifconfig 
cat /etc/rac_env_vars
ls -ltr /oradata
```

Also, please provide following from docker host:
```
docker inspect racnode1
docker network ls
docker network inspect rac_pub1_nw
docker network inspect rac_priv1_nw
docker volume inspect racstorage
```

**Comment by onkarnigam14** [comment]
ocker inspect racnode1
docker network ls
docker network inspect rac_pub1_nw
docker network inspect rac_priv1_nw
docker volume inspect racstorage


[root@ip-172-31-47-145 dockerfiles]# docker inspect racnode1
[
    {
        "Id": "6cc3ad01dda91e71937550b61277c5868eb6533cd2d6d1ac43b539cdbdbc5348",
        "Created": "2020-04-29T10:14:23.036377219Z",
        "Path": "/usr/sbin/oracleinit",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 12207,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2020-04-29T10:14:35.782301521Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "sha256:6fabe7985f25f3a595941b209bbb6da399e2ae05e8595280362de6bd945a7b19",
        "ResolvConfPath": "/var/lib/docker/containers/6cc3ad01dda91e71937550b61277c5868eb6533cd2d6d1ac43b539cdbdbc5348/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/6cc3ad01dda91e71937550b61277c5868eb6533cd2d6d1ac43b539cdbdbc5348/hostname",
        "HostsPath": "/opt/containers/rac_host_file",
        "LogPath": "/var/lib/docker/containers/6cc3ad01dda91e71937550b61277c5868eb6533cd2d6d1ac43b539cdbdbc5348/6cc3ad01dda91e71937550b61277c5868eb6533cd2d6d1ac43b539cdbdbc5348-json.log",
        "Name": "/racnode1",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": [
            "56551b656abc8d8be18ff48d927070141db76ac84de698db6dd087ce519ed4df"
        ],
        "HostConfig": {
            "Binds": [
                "/opt/containers/rac_host_file:/etc/hosts",
                "/opt/.secrets:/run/secrets",
                "racstorage:/oradata",
                "/sys/fs/cgroup:/sys/fs/cgroup:ro",
                "/boot:/boot:ro"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": [
                "SYS_NICE",
                "SYS_RESOURCE",
                "NET_ADMIN"
            ],
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [
                "example.com"
            ],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "Tmpfs": {
                "/dev/shm": "rw,exec,size=4G",
                "/run": ""
            },
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 95000,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": [
                {
                    "Name": "rtprio",
                    "Hard": 99,
                    "Soft": 99
                }
            ],
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/0d23d0cb241607a88dc6dccbd8cfd0acb1702ab22daafe8206b48a63adb8da33-init/diff:/var/lib/docker/overlay2/d1c32b4786ce36258d3b1bbeb25c496df3b200693c1282e88c9642d998ed1289/diff:/var/lib/docker/overlay2/7d5af1d2882b5f930baa25afaeca7da135542fa509b34dcca2d6166c890a44b0/diff:/var/lib/docker/overlay2/ab933d2b90709fa2ffbafe9edbc37374ee9ed3ad45450765a5ea07222893f650/diff:/var/lib/docker/overlay2/f5c3372c9e9ffe51f14ed69945f39a8e52ca82ffd228454e60c874826f1794f7/diff",
                "MergedDir": "/var/lib/docker/overlay2/0d23d0cb241607a88dc6dccbd8cfd0acb1702ab22daafe8206b48a63adb8da33/merged",
                "UpperDir": "/var/lib/docker/overlay2/0d23d0cb241607a88dc6dccbd8cfd0acb1702ab22daafe8206b48a63adb8da33/diff",
                "WorkDir": "/var/lib/docker/overlay2/0d23d0cb241607a88dc6dccbd8cfd0acb1702ab22daafe8206b48a63adb8da33/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/sys/fs/cgroup",
                "Destination": "/sys/fs/cgroup",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/boot",
                "Destination": "/boot",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "7b95d1d71887e2f16cef387482de84322d2b734c2eb19135e5537a7ca4b7db5e",
                "Source": "/var/lib/docker/volumes/7b95d1d71887e2f16cef387482de84322d2b734c2eb19135e5537a7ca4b7db5e/_data",
                "Destination": "/dev/shm",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "814a7d44faf8c46cfa8071096473da82f3b471642622d5d0712bce46f12e345b",
                "Source": "/var/lib/docker/volumes/814a7d44faf8c46cfa8071096473da82f3b471642622d5d0712bce46f12e345b/_data",
                "Destination": "/common_scripts",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/opt/containers/rac_host_file",
                "Destination": "/etc/hosts",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/.secrets",
                "Destination": "/run/secrets",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "racstorage",
                "Source": "/var/lib/docker/volumes/racstorage/_data",
                "Destination": "/oradata",
                "Driver": "local",
                "Mode": "z",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "racnode1",
            "Domainname": "",
            "User": "grid",
            "AttachStdin": true,
            "AttachStdout": true,
            "AttachStderr": true,
            "Tty": true,
            "OpenStdin": true,
            "StdinOnce": true,
            "Env": [
                "PRIV_HOSTNAME=racnode1-priv",
                "PUBLIC_IP=172.16.1.150",
                "SCAN_IP=172.16.1.70",
                "OP_TYPE=INSTALL",
                "ASM_DEVICE_LIST=/oradata/asm_disk01.img,/oradata/asm_disk02.img,/oradata/asm_disk03.img,/oradata/asm_disk04.img,/oradata/asm_disk05.img",
                "CMAN_IP=172.16.1.15",
                "PUBLIC_HOSTNAME=racnode1",
                "DOMAIN=example.com",
                "PWD_KEY=pwd.key",
                "PRIV_IP=192.168.17.150",
                "ASM_DISCOVERY_DIR=/oradata",
                "CMAN_HOSTNAME=racnode-cman1",
                "NODE_VIP=172.16.1.160",
                "VIP_HOSTNAME=racnode1-vip",
                "SCAN_NAME=racnode-scan",
                "COMMON_OS_PWD_FILE=common_os_pwdfile.enc",
                "PATH=/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "SETUP_LINUX_FILE=setupLinuxEnv.sh",
                "INSTALL_DIR=/opt/scripts",
                "GRID_BASE=/u01/app/grid",
                "GRID_HOME=/u01/app/19.3.0/grid",
                "INSTALL_FILE_1=LINUX.X64_193000_grid_home.zip",
                "GRID_INSTALL_RSP=gridsetup_19c.rsp",
                "GRID_SW_INSTALL_RSP=grid_sw_install_19c.rsp",
                "GRID_SETUP_FILE=setupGrid.sh",
                "FIXUP_PREQ_FILE=fixupPreq.sh",
                "INSTALL_GRID_BINARIES_FILE=installGridBinaries.sh",
                "INSTALL_GRID_PATCH=applyGridPatch.sh",
                "INVENTORY=/u01/app/oraInventory",
                "CONFIGGRID=configGrid.sh",
                "ADDNODE=AddNode.sh",
                "DELNODE=DelNode.sh",
                "ADDNODE_RSP=grid_addnode.rsp",
                "SETUPSSH=setupSSH.expect",
                "DOCKERORACLEINIT=dockeroracleinit",
                "GRID_USER_HOME=/home/grid",
                "SETUPGRIDENV=setupGridEnv.sh",
                "RESET_OS_PASSWORD=resetOSPassword.sh",
                "MULTI_NODE_INSTALL=MultiNodeInstall.py",
                "DB_BASE=/u01/app/oracle",
                "DB_HOME=/u01/app/oracle/product/19.3.0/dbhome_1",
                "INSTALL_FILE_2=LINUX.X64_193000_db_home.zip",
                "DB_INSTALL_RSP=db_sw_install_19c.rsp",
                "DBCA_RSP=dbca_19c.rsp",
                "DB_SETUP_FILE=setupDB.sh",
                "PWD_FILE=setPassword.sh",
                "RUN_FILE=runOracle.sh",
                "STOP_FILE=stopOracle.sh",
                "ENABLE_RAC_FILE=enableRAC.sh",
                "CHECK_DB_FILE=checkDBStatus.sh",
                "USER_SCRIPTS_FILE=runUserScripts.sh",
                "REMOTE_LISTENER_FILE=remoteListener.sh",
                "INSTALL_DB_BINARIES_FILE=installDBBinaries.sh",
                "GRID_HOME_CLEANUP=GridHomeCleanup.sh",
                "ORACLE_HOME_CLEANUP=OracleHomeCleanup.sh",
                "DB_USER=oracle",
                "GRID_USER=grid",
                "FUNCTIONS=functions.sh",
                "COMMON_SCRIPTS=/common_scripts",
                "CHECK_SPACE_FILE=checkSpace.sh",
                "RESET_FAILED_UNITS=resetFailedUnits.sh",
                "SET_CRONTAB=setCrontab.sh",
                "CRONTAB_ENTRY=crontabEntry",
                "EXPECT=/usr/bin/expect",
                "BIN=/usr/sbin",
                "container=true",
                "INSTALL_SCRIPTS=/opt/scripts/install",
                "SCRIPT_DIR=/opt/scripts/startup",
                "GRID_PATH=/u01/app/19.3.0/grid/bin:/u01/app/19.3.0/grid/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DB_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/bin:/u01/app/oracle/product/19.3.0/dbhome_1/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "GRID_LD_LIBRARY_PATH=/u01/app/19.3.0/grid/lib:/usr/lib:/lib",
                "DB_LD_LIBRARY_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/lib:/usr/lib:/lib"
            ],
            "Cmd": [
                "/usr/sbin/oracleinit"
            ],
            "Image": "oracle/database-rac:19.3.0",
            "Volumes": {
                "/common_scripts": {},
                "/dev/shm": {}
            },
            "WorkingDir": "/home/grid",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {}
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "d13b7f7a59436e428e26a935118f22793279fadf22e5c6e24bc6b75a15873918",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/d13b7f7a5943",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "85bccf19a8b4b9d2c33e10820b54b8ccc898b6cfcd144a91352049f1d3b5d115",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "1aaced4633f34148f66a5cc7c5ed39c31e53bec684eebd6ada9c5aff9315d693",
                    "EndpointID": "85bccf19a8b4b9d2c33e10820b54b8ccc898b6cfcd144a91352049f1d3b5d115",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02",
                    "DriverOpts": null
                }
            }
        }
    }
]
[root@ip-172-31-47-145 dockerfiles]#

===============================================================================================================================================

[root@ip-172-31-47-145 dockerfiles]# docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
1aaced4633f3        bridge              bridge              local
ea9deadeab86        host                host                local
e48bcf0ef8c2        none                null                local
912f30ef4bd5        rac_priv1_nw        bridge              local
c95017710999        rac_pub1_nw         bridge              local
[root@ip-172-31-47-145 dockerfiles]#


===============================================================================================================================================

[root@ip-172-31-47-145 dockerfiles]# docker network inspect rac_priv1_nw
[
    {
        "Name": "rac_priv1_nw",
        "Id": "912f30ef4bd51472033974078c6216a0381fa5e636c4ad8929815cfc8891adf9",
        "Created": "2020-04-28T08:12:48.860692222Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "192.168.17.0/24",
                    "Gateway": "192.168.17.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "e828fd36b9f5d25e111078a03909a09a067ce863644d5be28b1684483628b2ca": {
                "Name": "racnode-storage",
                "EndpointID": "52610775867c8eef3cdb177438dbe157954088a714efa6c42a8fb54be98a43e2",
                "MacAddress": "02:42:c0:a8:11:19",
                "IPv4Address": "192.168.17.25/24",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
[root@ip-172-31-47-145 dockerfiles]#



===============================================================================================================================================

[root@ip-172-31-47-145 dockerfiles]# docker network inspect rac_pub1_nw
[
    {
        "Name": "rac_pub1_nw",
        "Id": "c95017710999b8dbeae1985e66402784276c0bd10a8bf3e12e447d1bb64a1b00",
        "Created": "2020-04-28T08:12:18.197801951Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.16.1.0/24",
                    "Gateway": "172.16.1.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "e9564a213876ec288ecb6c37b753d7780e50da0739bea794c45ba6dd572a70a6": {
                "Name": "racnode-cman",
                "EndpointID": "f3edb4227d3d5ac5b826ab8303129c6c552ec274308e5b2d64826932e35b3594",
                "MacAddress": "02:42:ac:10:01:0f",
                "IPv4Address": "172.16.1.15/24",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
[root@ip-172-31-47-145 dockerfiles]#


===============================================================================================================================================

[root@ip-172-31-47-145 dockerfiles]# docker network inspect rac_pub1_nw
[
    {
        "Name": "rac_pub1_nw",
        "Id": "c95017710999b8dbeae1985e66402784276c0bd10a8bf3e12e447d1bb64a1b00",
        "Created": "2020-04-28T08:12:18.197801951Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.16.1.0/24",
                    "Gateway": "172.16.1.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "e9564a213876ec288ecb6c37b753d7780e50da0739bea794c45ba6dd572a70a6": {
                "Name": "racnode-cman",
                "EndpointID": "f3edb4227d3d5ac5b826ab8303129c6c552ec274308e5b2d64826932e35b3594",
                "MacAddress": "02:42:ac:10:01:0f",
                "IPv4Address": "172.16.1.15/24",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
[root@ip-172-31-47-145 dockerfiles]#
[root@ip-172-31-47-145 dockerfiles]#
[root@ip-172-31-47-145 dockerfiles]#
[root@ip-172-31-47-145 dockerfiles]# docker volume inspect racstorage
[
    {
        "CreatedAt": "2020-04-29T10:13:17Z",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/racstorage/_data",
        "Name": "racstorage",
        "Options": {
            "device": ":/oradata",
            "o": "addr=192.168.17.25,rw,bg,hard,tcp,vers=3,timeo=600,rsize=32768,wsize=32768,actimeo=0",
            "type": "nfs"
        },
        "Scope": "local"
    }
]
[root@ip-172-31-47-145 dockerfiles]#



===============================================================================================================================================

**Comment by onkarnigam14** [comment]
docker exec -i -t racnode1 /bin/bash
ifconfig 
cat /etc/rac_env_vars
ls -ltr /oradata

[root@ip-172-31-47-145 dockerfiles]#
[root@ip-172-31-47-145 dockerfiles]# docker exec -i -t racnode1 /bin/bash
[grid@racnode1 ~]$

========================================================================================================================

[root@ip-172-31-47-145 dockerfiles]# ifconfig
br-912f30ef4bd5: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.17.1  netmask 255.255.255.0  broadcast 192.168.17.255
        inet6 fe80::42:79ff:fe6d:6533  prefixlen 64  scopeid 0x20<link>
        ether 02:42:79:6d:65:33  txqueuelen 0  (Ethernet)
        RX packets 124456  bytes 7314202 (6.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 143744  bytes 1252447276 (1.1 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

br-c95017710999: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.1.1  netmask 255.255.255.0  broadcast 172.16.1.255
        inet6 fe80::42:86ff:feeb:3eae  prefixlen 64  scopeid 0x20<link>
        ether 02:42:86:eb:3e:ae  txqueuelen 0  (Ethernet)
        RX packets 5  bytes 440 (440.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5  bytes 440 (440.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        inet6 fe80::42:33ff:fe9d:6eda  prefixlen 64  scopeid 0x20<link>
        ether 02:42:33:9d:6e:da  txqueuelen 0  (Ethernet)
        RX packets 124456  bytes 7314202 (6.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 143744  bytes 1252447276 (1.1 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.31.47.145  netmask 255.255.240.0  broadcast 172.31.47.255
        inet6 fe80::f1:10ff:fe2f:6bd6  prefixlen 64  scopeid 0x20<link>
        ether 02:f1:10:2f:6b:d6  txqueuelen 1000  (Ethernet)
        RX packets 6009901  bytes 8681755713 (8.0 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1980307  bytes 117138877 (111.7 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 5  bytes 440 (440.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5  bytes 440 (440.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth8445ae6: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::c4ca:aff:fe28:a888  prefixlen 64  scopeid 0x20<link>
        ether c6:ca:0a:28:a8:88  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 14  bytes 1116 (1.0 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

veth9b0d960: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::a809:6eff:feac:57f5  prefixlen 64  scopeid 0x20<link>
        ether aa:09:6e:ac:57:f5  txqueuelen 0  (Ethernet)
        RX packets 158  bytes 26744 (26.1 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 183  bytes 34666 (33.8 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

vethe6f06c9: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::d46f:d9ff:fe3e:28b0  prefixlen 64  scopeid 0x20<link>
        ether d6:6f:d9:3e:28:b0  txqueuelen 0  (Ethernet)
        RX packets 4  bytes 240 (240.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 17  bytes 1417 (1.3 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[root@ip-172-31-47-145 dockerfiles]#

=============================================================================================================================
[grid@racnode1 ~]$
[grid@racnode1 ~]$ ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.2  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:ac:11:00:02  txqueuelen 0  (Ethernet)
        RX packets 18  bytes 1487 (1.4 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 4  bytes 240 (240.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[grid@racnode1 ~]$

========================================================================================================================

[grid@racnode1 ~]$ cat /etc/rac_env_vars
export PATH=/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
export HOSTNAME=racnode1
export TERM=xterm
export PRIV_HOSTNAME=racnode1-priv
export PUBLIC_IP=172.16.1.150
export SCAN_IP=172.16.1.70
export OP_TYPE=INSTALL
export ASM_DEVICE_LIST=/oradata/asm_disk01.img,/oradata/asm_disk02.img,/oradata/asm_disk03.img,/oradata/asm_disk04.img,/oradata/asm_disk05.img
export CMAN_IP=172.16.1.15
export PUBLIC_HOSTNAME=racnode1
export DOMAIN=example.com
export PWD_KEY=pwd.key
export PRIV_IP=192.168.17.150
export ASM_DISCOVERY_DIR=/oradata
export CMAN_HOSTNAME=racnode-cman1
export NODE_VIP=172.16.1.160
export VIP_HOSTNAME=racnode1-vip
export SCAN_NAME=racnode-scan
export COMMON_OS_PWD_FILE=common_os_pwdfile.enc
export SETUP_LINUX_FILE=setupLinuxEnv.sh
export INSTALL_DIR=/opt/scripts
export GRID_BASE=/u01/app/grid
export GRID_HOME=/u01/app/19.3.0/grid
export INSTALL_FILE_1=LINUX.X64_193000_grid_home.zip
export GRID_INSTALL_RSP=gridsetup_19c.rsp
export GRID_SW_INSTALL_RSP=grid_sw_install_19c.rsp
export GRID_SETUP_FILE=setupGrid.sh
export FIXUP_PREQ_FILE=fixupPreq.sh
export INSTALL_GRID_BINARIES_FILE=installGridBinaries.sh
export INSTALL_GRID_PATCH=applyGridPatch.sh
export INVENTORY=/u01/app/oraInventory
export CONFIGGRID=configGrid.sh
export ADDNODE=AddNode.sh
export DELNODE=DelNode.sh
export ADDNODE_RSP=grid_addnode.rsp
export SETUPSSH=setupSSH.expect
export DOCKERORACLEINIT=dockeroracleinit
export GRID_USER_HOME=/home/grid
export SETUPGRIDENV=setupGridEnv.sh
export RESET_OS_PASSWORD=resetOSPassword.sh
export MULTI_NODE_INSTALL=MultiNodeInstall.py
export DB_BASE=/u01/app/oracle
export DB_HOME=/u01/app/oracle/product/19.3.0/dbhome_1
export INSTALL_FILE_2=LINUX.X64_193000_db_home.zip
export DB_INSTALL_RSP=db_sw_install_19c.rsp
export DBCA_RSP=dbca_19c.rsp
export DB_SETUP_FILE=setupDB.sh
export PWD_FILE=setPassword.sh
export RUN_FILE=runOracle.sh
export STOP_FILE=stopOracle.sh
export ENABLE_RAC_FILE=enableRAC.sh
export CHECK_DB_FILE=checkDBStatus.sh
export USER_SCRIPTS_FILE=runUserScripts.sh
export REMOTE_LISTENER_FILE=remoteListener.sh
export INSTALL_DB_BINARIES_FILE=installDBBinaries.sh
export GRID_HOME_CLEANUP=GridHomeCleanup.sh
export ORACLE_HOME_CLEANUP=OracleHomeCleanup.sh
export DB_USER=oracle
export GRID_USER=grid
export FUNCTIONS=functions.sh
export COMMON_SCRIPTS=/common_scripts
export CHECK_SPACE_FILE=checkSpace.sh
export RESET_FAILED_UNITS=resetFailedUnits.sh
export SET_CRONTAB=setCrontab.sh
export CRONTAB_ENTRY=crontabEntry
export EXPECT=/usr/bin/expect
export BIN=/usr/sbin
export container=true
export INSTALL_SCRIPTS=/opt/scripts/install
export SCRIPT_DIR=/opt/scripts/startup
export GRID_PATH=/u01/app/19.3.0/grid/bin:/u01/app/19.3.0/grid/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
export DB_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/bin:/u01/app/oracle/product/19.3.0/dbhome_1/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
export GRID_LD_LIBRARY_PATH=/u01/app/19.3.0/grid/lib:/usr/lib:/lib
export DB_LD_LIBRARY_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/lib:/usr/lib:/lib
export HOME=/home/grid
[grid@racnode1 ~]$

========================================================================================================================

[grid@racnode1 ~]$ ls -ltr /oradata
total 52428820
-rw-rw----. 1 grid asmadmin 10737418240 Apr 28 08:13 asm_disk01.img
-rw-rw----. 1 grid asmadmin 10737418240 Apr 28 08:15 asm_disk02.img
-rw-rw----. 1 grid asmadmin 10737418240 Apr 28 08:16 asm_disk03.img
-rw-rw----. 1 grid asmadmin 10737418240 Apr 28 08:18 asm_disk04.img
-rw-rw----. 1 grid asmadmin 10737418240 Apr 28 08:19 asm_disk05.img
[grid@racnode1 ~]$



========================================================================================================================

**Comment by onkarnigam14** [comment]
When I have login to RAC container that i have created from the image : oracle/database-rac         19.3.0              6fabe7985f25 

it is throwing the error when I am running the command in the container : oracleinit

[grid@racnode1 ~]$ oracleinit
ENABLE_RAC_FILE=enableRAC.sh
HOSTNAME=racnode1
ASM_DEVICE_LIST=/oradata/asm_disk01.img,/oradata/asm_disk02.img,/oradata/asm_disk03.img,/oradata/asm_disk04.img,/oradata/asm_disk05.img
REMOTE_LISTENER_FILE=remoteListener.sh
DB_BASE=/u01/app/oracle
TERM=xterm
DB_INSTALL_RSP=db_sw_install_19c.rsp
INSTALL_DIR=/opt/scripts
DB_HOME=/u01/app/oracle/product/19.3.0/dbhome_1
ORACLE_HOME_CLEANUP=OracleHomeCleanup.sh
INSTALL_GRID_BINARIES_FILE=installGridBinaries.sh
DBCA_RSP=dbca_19c.rsp
GRID_BASE=/u01/app/grid
OP_TYPE=INSTALL
STOP_FILE=stopOracle.sh
PWD_KEY=pwd.key
SCAN_IP=172.16.1.70
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=01;05;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.axv=01;35:*.anx=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=01;36:*.au=01;36:*.flac=01;36:*.mid=01;36:*.midi=01;36:*.mka=01;36:*.mp3=01;36:*.mpc=01;36:*.ogg=01;36:*.ra=01;36:*.wav=01;36:*.axa=01;36:*.oga=01;36:*.spx=01;36:*.xspf=01;36:
CONFIGGRID=configGrid.sh
GRID_INSTALL_RSP=gridsetup_19c.rsp
GRID_LD_LIBRARY_PATH=/u01/app/19.3.0/grid/lib:/usr/lib:/lib
EXPECT=/usr/bin/expect
INSTALL_FILE_2=LINUX.X64_193000_db_home.zip
SETUP_LINUX_FILE=setupLinuxEnv.sh
PUBLIC_HOSTNAME=racnode1
PRIV_HOSTNAME=racnode1-priv
CHECK_DB_FILE=checkDBStatus.sh
INSTALL_FILE_1=LINUX.X64_193000_grid_home.zip
VIP_HOSTNAME=racnode1-vip
DB_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/bin:/u01/app/oracle/product/19.3.0/dbhome_1/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
CHECK_SPACE_FILE=checkSpace.sh
FUNCTIONS=functions.sh
GRID_SW_INSTALL_RSP=grid_sw_install_19c.rsp
PATH=/u01/app/19.3.0/grid/bin:/u01/app/19.3.0/grid/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
GRID_HOME_CLEANUP=GridHomeCleanup.sh
CMAN_HOSTNAME=racnode-cman1
CRONTAB_ENTRY=crontabEntry
PWD=/home/grid
ADDNODE=AddNode.sh
RESET_OS_PASSWORD=resetOSPassword.sh
SCAN_NAME=racnode-scan
DOCKERORACLEINIT=dockeroracleinit
DOMAIN=example.com
PWD_FILE=setPassword.sh
SET_CRONTAB=setCrontab.sh
RUN_FILE=runOracle.sh
SCRIPT_DIR=/opt/scripts/startup
PRIV_IP=192.168.17.150
PUBLIC_IP=172.16.1.150
GRID_PATH=/u01/app/19.3.0/grid/bin:/u01/app/19.3.0/grid/OPatch/:/usr/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
COMMON_SCRIPTS=/common_scripts
FIXUP_PREQ_FILE=fixupPreq.sh
ASM_DISCOVERY_DIR=/oradata
GRID_USER_HOME=/home/grid
INVENTORY=/u01/app/oraInventory
SHLVL=1
HOME=/home/grid
USER_SCRIPTS_FILE=runUserScripts.sh
GRID_USER=grid
DB_LD_LIBRARY_PATH=/u01/app/oracle/product/19.3.0/dbhome_1/lib:/usr/lib:/lib
MULTI_NODE_INSTALL=MultiNodeInstall.py
GRID_HOME=/u01/app/19.3.0/grid
BIN=/usr/sbin
SETUPSSH=setupSSH.expect
ADDNODE_RSP=grid_addnode.rsp
DB_USER=oracle
INSTALL_GRID_PATCH=applyGridPatch.sh
SETUPGRIDENV=setupGridEnv.sh
DB_SETUP_FILE=setupDB.sh
COMMON_OS_PWD_FILE=common_os_pwdfile.enc
RESET_FAILED_UNITS=resetFailedUnits.sh
INSTALL_SCRIPTS=/opt/scripts/install
NODE_VIP=172.16.1.160
container=true
INSTALL_DB_BINARIES_FILE=installDBBinaries.sh
DELNODE=DelNode.sh
GRID_SETUP_FILE=setupGrid.sh
CMAN_IP=172.16.1.15
_=/usr/sbin/oracleinit
Trying to run as user instance, but $XDG_RUNTIME_DIR is not set.
[grid@racnode1 ~]$
[grid@racnode1 ~]$
[grid@racnode1 ~]$

**Comment by psaini79** [comment]
@onkarnigam14 

Here is the problem , you have not disconnected the default bridge and not assigned the bridge rac_pub1_nw to the container. You are still on 172.17.0.0/16 network.

```
[grid@racnode1 ~]$ ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST> mtu 1500
inet 172.17.0.2 netmask 255.255.0.0 broadcast 172.17.255.255
```
However, you have passed the PUBLIC_IP=172.16.1.150 to container so because of network mismatch it is not working. Please do following:

```
docker network disconnect bridge racnode1
docker network connect rac_pub1_nw --ip 172.16.1.150 racnode1
docker network connect rac_priv1_nw --ip 192.168.17.150  racnode1
```
docker start racnode1
docker logs -f racnode1
Let  me know if you are still unable to proceed.

**Comment by onkarnigam14** [comment]
Thanks I was able to run it 
Now I am facing an issue of racnode2 :

[root@ip-172-31-47-145 ~]# docker start racnode2
Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"process_linux.go:367: setting cgroup config for procHooks process caused \\\"failed to write 95000 to cpu.rt_runtime_us: write /sys/fs/cgroup/cpu,cpuacct/system.slice/docker-22622cde65973d14f68648f212d99f2d3bf6c8c1e58e8c8d64322def3d52380b.scope/cpu.rt_runtime_us: invalid argument\\\"\"": unknown
Error: failed to start containers: racnode2

Can you help me with that 


though I have allocated more memory to it 
and also CPU 
32 VCPU and 128GB memory I have taken 

so how can I resolve this error

**Comment by onkarnigam14** [comment]
I removed the line :   --cpu-rt-runtime=95000 \ from the container and resolved the issue 

But it trigerred a new issue : 
The scripts can be executed in parallel on all the nodes.

Successfully Setup Software.
04-30-2020 07:55:57 UTC :  : Node Addition performed. removing Responsefile
04-30-2020 07:55:57 UTC :  : Running root.sh on node racnode2
04-30-2020 07:55:57 UTC :  : Nodes in the cluster racnode2
04-30-2020 07:57:47 UTC :  : Checking Cluster
04-30-2020 07:57:47 UTC :  : Cluster Check passed
04-30-2020 07:57:47 UTC :  : Cluster Check went fine
04-30-2020 07:57:47 UTC : : CRSD Check failed!
04-30-2020 07:57:47 UTC : : Error has occurred in Grid Setup, Please verify!
[grid@racnode2 ~]$

 please help me out

**Comment by onkarnigam14** [comment]
HI By mistake i removed my container , and now the container is not coming up , because of some issue 
though everything is same 
04-30-2020 11:09:34 UTC :  : Running DB creation
04-30-2020 11:09:37 UTC :  : Checking DB status
04-30-2020 11:09:37 UTC : : ORCLCDB is not up and running on racnode1
04-30-2020 11:09:37 UTC : : Error has occurred in Grid Setup, Please verify!

**Comment by onkarnigam14** [comment]
I have performed the below mentioned steps but then also the issue was not resolved 

 /opt/.secrets
docker exec racnode1 /bin/sh -c 'ls -ltr /run/secrets'
rm -f /opt/.secrets/*
docker exec racnode1 /bin/sh -c 'ls -ltr /run/secrets' Note: No file must be there on racnode1 under /run/secrets
vi /opt/.secrets/common_os_pwdfile and seed your password. It must be as per Oracle standard.
openssl rand -hex 64 -out /opt/.secrets/pwd.key
openssl enc -aes-256-cbc -salt -in /opt/.secrets/common_os_pwdfile -out /opt/.secrets/common_os_pwdfile.enc -pass file:/opt/.secrets/pwd.key
docker exec racnode1 /bin/sh -c 'ls -ltr /run/secrets' Note: New files must be there
rm -f /opt/.secrets/common_os_pwdfile
Execute following steps: docker exec -i -t racnode1 /bin/bash sudo /bin/bash sh -x /opt/scripts/startup/runOracle.sh

**Comment by onkarnigam14** [comment]
[root@ip-172-31-47-145 opt]# docker exec -i -t -u root racnode1 /bin/bash
OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused "chdir to cwd (\"/home/grid\") set in config.json failed: permission denied": unknown
[root@ip-172-31-47-145 opt]#

**Comment by psaini79** [comment]
@onkarnigam14 

I am surprised that 2nd container is unable to get RT quota. I can see you have given following allocation to dockerd:

```
/usr/bin/dockerd --cpu-rt-runtime=950000 --cpu-rt-period=1000000 --exec-opt=native.cgroupdriver=systemd
```
However, please do following and try again on 19.3 docker ol and let's use default cgroupfs drive:
* If Oracle docker-engine version >= 19.3
  * Edit /usr/lib/systemd/system/docker.service and add --cpu-rt-runtime=950000 additional parameters in [Service] section for dockerd daemon.
```
ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock --cpu-rt-runtime=950000
```
Retstart systemd
```
systemctl daemon-reload
systemctl restart docker
```
**Note** : I will update the same on README.md.

Since you have deleted the racnode1 container and recreated it, did you recreate storage container and deleted the files from /oradata so that new dummy devices can be created using  again for ASM storage?  If not, I request to delete racnode1 and racstorage container and cleanup the volume where /oradata was mounted .

Recreate racstorage and once the creation is finished then recreate racnode1.


To login as root inside the container, please do following:
docker exec -i -t racnode1 /bin/bash
sudo /bin/bash

**Comment by onkarnigam14** [comment]
Facing issue in racnode2 cluster : 

CVU operation performed:      stage -pre nodeadd
Date:                         Apr 30, 2020 4:42:17 PM
CVU home:                     /u01/app/19.3.0/grid/
User:                         grid
04-30-2020 16:43:06 UTC :  : CVU Checks are ignored as IGNORE_CVU_CHECKS set to true. It is recommended to set IGNORE_CVU_CHECKS to false and meet all the cvu checks requirement. RAC installation might fail, if there are failed cvu checks.
04-30-2020 16:43:06 UTC :  : Running Node Addition and cluvfy test for node racnode2
04-30-2020 16:43:06 UTC :  : Copying /tmp/grid_addnode.rsp on remote node racnode1
04-30-2020 16:43:06 UTC :  : Running GridSetup.sh on racnode1 to add the node to existing cluster
04-30-2020 16:43:50 UTC :  : Node Addition performed. removing Responsefile
04-30-2020 16:43:50 UTC :  : Running root.sh on node racnode2
04-30-2020 16:43:50 UTC :  : Nodes in the cluster racnode2
04-30-2020 16:45:24 UTC :  : Checking Cluster
04-30-2020 16:45:24 UTC :  : Cluster Check passed
04-30-2020 16:45:24 UTC :  : Cluster Check went fine
04-30-2020 16:45:24 UTC : : CRSD Check failed!
04-30-2020 16:45:24 UTC : : Error has occurred in Grid Setup, Please verify!

**Comment by psaini79** [comment]
@onkarnigam14 

Please upload the logs of grid of racnode2 and racnode1.
Login to racnode1 and racnode2
cd /u01/app/grid/
tar -cvzf logs_<node_name>.log  diag

Upload the logs.

**Comment by onkarnigam14** [comment]
Ok , how to upload the logs from inside a container

**Comment by onkarnigam14** [comment]
[racnode11.log](https://github.com/oracle/docker-images/files/4562236/racnode11.log)
[racnode2.log](https://github.com/oracle/docker-images/files/4562240/racnode2.log)

**Comment by onkarnigam14** [comment]
I have uploaded the files, thanks

**Comment by psaini79** [comment]
@onkarnigam14 
I am unable to open logs and change directories. Please execute following steps:

Login to docker host and execute following steps:
```
docker exec -i -t racnode1 /bin/bash
cd /u01/app/grid/
tar -cvzf logs_<node_name>.tgz  diag
docker cp racnode1:/u01/app/grid/logs_<node_name>.tgz .
```
Do the same with racnode2 logs and upload the logs.

**Comment by onkarnigam14** [comment]
Please find the below logs : 
Racnode1
[logs.zip](https://github.com/oracle/docker-images/files/4562395/logs.zip)

I have uploaded both the logs file

**Comment by psaini79** [comment]
@onkarnigam14 ,

I looked at the logs and It seems cssd is unable to come up in RT mode. 
```
2020-05-01 03:27:32.640 [OCSSD(14585)]CRS-1726: Process failed to run in real-time priority. Details at (:CLSN00143:) in /u01/app/grid/diag/crs/racnode2/crs/trace/ocssd.trc.
2020-05-01 03:27:32.640 [OCSSD(14585)]CRS-1656: The CSS daemon is terminating due to a fatal error; Details at (:CSSSC00011:) in 
```

Please paste the command you used to create the racnode2 container.

**Comment by onkarnigam14** [comment]
NTP service is not installed in my main OS, so because of that I am facing the issue.

**Comment by onkarnigam14** [comment]
docker create -t -i \
  --hostname racnode2 \
  --volume /boot:/boot:ro \
  --volume /dev/shm \
  --tmpfs /dev/shm:rw,exec,size=4G \
  --volume /opt/containers/rac_host_file:/etc/hosts  \
  --volume /opt/.secrets:/run/secrets \
  --dns-search=example.com \
  --privileged=false \
  --volume racstorage:/oradata \
  --cap-add=SYS_NICE \
  --cap-add=SYS_RESOURCE \
  --cap-add=NET_ADMIN \
  -e NODE_VIP=172.16.1.161  \
  -e VIP_HOSTNAME=racnode2-vip  \
  -e PRIV_IP=192.168.17.151  \
  -e PRIV_HOSTNAME=racnode2-priv \
  -e PUBLIC_IP=172.16.1.151 \
  -e PUBLIC_HOSTNAME=racnode2  \
  -e EXISTING_CLS_NODES=racnode1 \
  -e SCAN_NAME=racnode-scan \
  -e SCAN_IP=172.16.1.70  \
  -e OP_TYPE=INSTALL \
  -e DOMAIN=example.com \
  -e ASM_DISCOVERY_DIR=/oradata \
  -e ASM_DEVICE_LIST=/oradata/asm_disk01.img,/oradata/asm_disk02.img,/oradata/asm_disk03.img,/oradata/asm_disk04.img,/oradata/asm_disk05.img  \
  -e ORACLE_SID=ORCLCDB \
  -e OP_TYPE=ADDNODE \
  -e CMAN_HOSTNAME=racnode-cman1 \
  -e CMAN_IP=172.16.1.15 \
  -e COMMON_OS_PWD_FILE=common_os_pwdfile.enc \
  -e PWD_KEY=pwd.key \
  --tmpfs=/run -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
  --ulimit rtprio=99  \
  --name racnode2 \
  oracle/database-rac:19.3.0

**Comment by psaini79** [comment]
@onkarnigam14 

This is expected as you have removed cpu runtime allocation clause to container. If you check the README.md, the following clause is there but your contianer creation command does not have it:

--cpu-rt-runtime=95000 

Please create the container with correct parameters, re-run the setup and it should work.

**Comment by onkarnigam14** [comment]
I tried that also but it failed

**Comment by psaini79** [comment]
@onkarnigam14 

Can you give me the logs from racnode2
```
docker exec -i -t racnode2 /bin/bash
cd /u01/app/grid/
tar -cvzf logs_<node_name>.tgz  diag
docker cp racnode2:/u01/app/grid/logs_<node_name>.tgz .
```

Also, please let me know the sequence of steps.

**Comment by onkarnigam14** [comment]
[grid@racnode2 ~]$ crsctl check cluster -all
**************************************************************
racnode1:
CRS-4537: Cluster Ready Services is online
CRS-4529: Cluster Synchronization Services is online
CRS-4533: Event Manager is online
**************************************************************
racnode2:
CRS-4535: Cannot communicate with Cluster Ready Services
CRS-4530: Communications failure contacting Cluster Synchronization Services daemon
CRS-4534: Cannot communicate with Event Manager
**************************************************************

[logs_racnode2.zip](https://github.com/oracle/docker-images/files/4567134/logs_racnode2.zip)

**Comment by psaini79** [comment]
@onkarnigam14 

Please provide following output from docker host:
```
route -n
docker network ls
```

Login to racnode2 and racnode1 container and provide following output:
```
ifconfig
```

**Comment by onkarnigam14** [comment]
[grid@racnode2 ~]$ ifconfig -a
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.17.151  netmask 255.255.255.0  broadcast 192.168.17.255
        ether 02:42:c0:a8:11:97  txqueuelen 0  (Ethernet)
        RX packets 19685  bytes 3280216 (3.1 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 19628  bytes 3082707 (2.9 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.1.151  netmask 255.255.255.0  broadcast 172.16.1.255
        ether 02:42:ac:10:01:97  txqueuelen 0  (Ethernet)
        RX packets 23857  bytes 51352330 (48.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 21669  bytes 5134910 (4.8 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 40166  bytes 6758386 (6.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 40166  bytes 6758386 (6.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[grid@racnode2 ~]$





[grid@racnode1 ~]$ ifconfig -a
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.17.150  netmask 255.255.255.0  broadcast 192.168.17.255
        ether 02:42:c0:a8:11:96  txqueuelen 0  (Ethernet)
        RX packets 20066  bytes 3133869 (2.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 20690  bytes 3415201 (3.2 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0:1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.15.81  netmask 255.255.224.0  broadcast 169.254.31.255
        ether 02:42:c0:a8:11:96  txqueuelen 0  (Ethernet)

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.1.150  netmask 255.255.255.0  broadcast 172.16.1.255
        ether 02:42:ac:10:01:96  txqueuelen 0  (Ethernet)
        RX packets 21818  bytes 5153771 (4.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 24085  bytes 51383704 (49.0 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1:1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.1.160  netmask 255.255.255.0  broadcast 172.16.1.255
        ether 02:42:ac:10:01:96  txqueuelen 0  (Ethernet)

eth1:2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.1.70  netmask 255.255.255.0  broadcast 172.16.1.255
        ether 02:42:ac:10:01:96  txqueuelen 0  (Ethernet)

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 317141  bytes 961526483 (916.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 317141  bytes 961526483 (916.9 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[grid@racnode1 ~]$
[logs_route.txt](https://github.com/oracle/docker-images/files/4567346/logs_route.txt)

**Comment by onkarnigam14** [comment]
[root@ip-172-31-42-105 ~]# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         172.31.32.1     0.0.0.0         UG    0      0        0 eth0
169.254.0.0     0.0.0.0         255.255.0.0     U     1002   0        0 eth0
172.16.1.0      0.0.0.0         255.255.255.0   U     0      0        0 br-ae931e39c283
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
172.31.32.0     0.0.0.0         255.255.240.0   U     0      0        0 eth0
192.168.17.0    0.0.0.0         255.255.255.0   U     0      0        0 br-23492cf29ac5
[root@ip-172-31-42-105 ~]#

**Comment by psaini79** [comment]
@onkarnigam14 

Your network configuration seems to be right. Can you please give me following logs from racnode1:
```
docker exec -i -t racnode1 /bin/bash
cd /u01/app/grid/
tar -cvzf logs_<node_name>.tgz  diag
docker cp racnode1:/u01/app/grid/logs_<node_name>.tgz .
```
You have already given me logs from node2. Please share above logs from node1. Let me look at node1 logs and it is possible we might need to enable tracing to debug the issue.

I tested the same conf on my machine with same Oracle Linux version, oracle docker container runtime  and kernel  version to reproduce the issue but I did not see any issue.

**Comment by onkarnigam14** [comment]
Everytime I shutdown the aws instance, when I bring it back it does have issue how to resolve that.  What commands o have to issue when I have to stop the container and bring it back up i.e racnode1

**Comment by onkarnigam14** [comment]
[logs_racnode1.zip](https://github.com/oracle/docker-images/files/4569502/logs_racnode1.zip)

[grid@racnode1 ~]$ oifcfg getif
eth0  192.168.17.0  global  cluster_interconnect,asm
eth1  172.16.1.0  global  public
[grid@racnode1 ~]$

**Comment by onkarnigam14** [comment]
If you want to connect to my instance , you can do it , I have hosted it on AWS :

Public IP : 13.127.168.40

Please share your mail id will share key with you

**Comment by onkarnigam14** [comment]
I am sharing the key, once you have copied it, please let me know, will remove it.
[oracle.zip](https://github.com/oracle/docker-images/files/4569539/oracle.zip)

**Comment by onkarnigam14** [comment]
Hi Good morning, did you check my logs

**Comment by psaini79** [comment]
@onkarnigam14,

Sorry for the delayed reply.  I looked  at the logs and RAC must be up and running on both the nodes. Can you try to run reastart? both the containers again, start Grid Infrastructure and check the following:
```
crsctl check crs
crsctl check cluster
```
Also, you might need to run root.sh again on racnode2 as your setup has not completed.

I also recommend using **Oracle RAC on Docker** on KVM or virtual box on-prem or in Oracle Cloud ( dev and test) populated with OEL 7.x and UEK5 for further assistance, because Oracle RAC is only supported in the Oracle Cloud: https://www.oracle.com/technetwork/database/options/clustering/overview/rac-cloud-support-2843861.pdf

**Comment by psaini79** [comment]
@onkarnigam14 

Please let me know if you tried on OL  7.x? If you are not seeing any issue please close this thread.

**Comment by onkarnigam14** [comment]
Yeah thanks i.will close the thread
